cashu/wallet/api/router.py

```diff
@@ -139,6 +140,58 @@ async def invoice(
     return
 
 
+@router.post(
+    "/swap",
+    name="Multi-mint swaps",
+    summary="Swap funds between mints.",
+    response_model=SwapResponse,
+)
+async def swap(
+    amount: int = Query(default=..., description="Amount to swap between mints"),
+    outgoing_mint: str = Query(default=..., description="URL of outgoing mint"),
+    incoming_mint: str = Query(default=..., description="URL of incoming mint"),
+):
+    if not settings.lightning:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="this mint does not support lightning.",
+        )
+    if outgoing_mint in incoming_mint:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="mints for swap have to be different",
+        )
+    global wallet
+    # request invoice from incoming mint
+    wallet = await load_mint(wallet, incoming_mint)
+    invoice = await wallet.request_mint(amount)
+    # pay invoice from outgoing mint
+    wallet = await load_mint(wallet, outgoing_mint)
+    await wallet.load_proofs()
```

Comment:
Question: If `wallet` is a **global**, and the mint is changed here, during this time any other requests to the wallet API could accidentally get the outgoing mint as the loaded mint?

Additionally, another request could itself change the mint? i.e. if I made two rapid requests to `/swap`, there could be four different mints involved and the swaps could mess each-other up?